core/src/saros/communication/connection/ConnectionHandler.java

```diff
@@ -113,6 +113,16 @@ public String getConnectionID() {
     return jid.toString();
   }
 
+  /**
+   * Returns the local JID for the current connection.
+   *
+   * @return the local JID for the current connection or <code>null</code> if no connection is
+   *     established or the connection has no jid
+   */
+  public JID getLocalJID() {
```

Comment:
This basically violates the design. This class should hide the any internals. What is the problem to use getConnectID and handle this in the affected places ?

Reply:
I know, but as the whole UI is currently violating the whole design I added this.
For example `OpenChatAction` is comparing JIDs, using `getConnectID()` this would lead to jid -> string -> jid conversions without a benefit because in the end everything its still limited to using jids.

I could add a `@deprecated` remark, but I think that this is superfluous because all JID usage in UI should be deprecated.

So yes this can look like a little step back, but from my point of view its better to do so for now. My current approach is to focus on moving smack stuff to core and than removing all this JID refs. Are you ok with this?